src/backy/backends/chunked/store.py

```diff
@@ -24,92 +21,53 @@ class Store(object):
     # wanting to perform new backups.
     force_writes = False
 
-    path: str
-    users: list[File]
-    seen_forced: set[str]
-    known: set[str]
+    path: Path
+    seen_forced: set[Hash]
+    known: set[Hash]
     log: BoundLogger
 
-    def __init__(self, path, log):
+    def __init__(self, path: Path, log: BoundLogger):
         self.path = path
-        self.users = []
         self.seen_forced = set()
-        self.known = set()
         self.log = log.bind(subsystem="chunked-store")
         for x in range(256):
-            subdir = os.path.join(self.path, f"{x:02x}")
-            if not os.path.exists(subdir):
-                try:
-                    os.makedirs(subdir)
-                except FileExistsError:
-                    pass
-        if not os.path.exists(os.path.join(self.path, "store")):
+            subdir = self.path / f"{x:02x}"
+            subdir.mkdir(parents=True, exist_ok=True)
+        if not self.path.joinpath("store").exists():
             self.convert_to_v2()
 
-        for c in glob.iglob(os.path.join(self.path, "*/*.chunk.lzo")):
-            hash = os.path.basename(c).replace(".chunk.lzo", "")
-            self.known.add(hash)
+        self.known = set(self.ls())
```

Comment:
There's a nice optimization here: `known` doesn't have to be `all known`, but `yes, we know this exists` so we can use it as a cache.

However, the `ls` is really only relevant for purging, so let's limit the `all` semantics to the purge command.

Reply:
That also means, chunk.py:132 needs to actually test for existence (and can cache it in the known set if it wants to)